src/basic/time-util.c

```diff
@@ -282,8 +282,11 @@ static char *format_timestamp_internal(
                 return NULL; /* Timestamp is unset */
 
         /* Let's not format times with years > 9999 */
-        if (t > USEC_TIMESTAMP_FORMATTABLE_MAX)
-                return NULL;
+        if (t > USEC_TIMESTAMP_FORMATTABLE_MAX) {
+                assert(l >= strlen("--- XXXX-XX-XX XX:XX:XX"));
```

Comment:
STRLEN() rather than strlen() please.

also this is off by one, no space for the trailing NUL is included

Reply:
Off-by-one fixed.

> STRLEN() rather than strlen() please

The compilers (gcc and clang both) are generally able to optimize strlen away to a constant, even at `-O0`. The reason we have STRLEN is that they still created a VLA despite optimizing the size to a constant. So we should use STRLEN in variable declarations, but otherwise plain strlen is completely fine.